### PR TITLE
plan: fix a bug about prefix index.

### PR DESCRIPTION
**Pull request by hanfei1991**
For index(c(3)), we should build range for ne condition. For c != 'abc', the range will be (-inf, abc], [abc, + inf), it will read abc two times.
@coocood @shenli @zimulala PTAL

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

File: plan/statistics/statistics.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ const (
 	// between condition selects 1/40 of total rows.
 	pseudoRowCount    = 10000000
 	pseudoEqualRate   = 1000
-	pseudoLessRate    = 3
+	pseudoLessRate    = 30
```
Comment (coocood) [inline plan/statistics/statistics.go]: This rate number indicates a less condition has 1/30 possibility to pass, which does not reflect the fact.

Comment (lamxTyler) [inline plan/statistics/statistics.go]: Please update the comment.

**Comment by coocood** [comment]
If we adjust the cost model drastically, we are at a big risk to make previously working application fail when they upgrade TiDB.

**Comment by shenli** [comment]
@hanfei1991 Any update?

**Comment by coocood** [comment]
Please fix the prefix index bug without adjusting CBO params, then we can merge it soon.

**Comment by hanfei1991** [comment]
@shenli @coocood @zimulala PTAL

**Comment by shenli** [comment]
@coocood PTAL

**Pull request merged**